test/e2e/data/infrastructure-aws/kustomize_sources/machine-pool/spot-instance-machine-pool.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachinePool
+metadata:
+  name: ${CLUSTER_NAME}-mp-1
+spec:
+  clusterName: ${CLUSTER_NAME}
+  replicas: ${WORKER_MACHINE_COUNT}
+  template:
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+          kind: KubeadmConfig
+          name: ${CLUSTER_NAME}-mp-1
+      clusterName: ${CLUSTER_NAME}
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: AWSMachinePool
+        name: ${CLUSTER_NAME}-mp-1
+      version: ${KUBERNETES_VERSION}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AWSMachinePool
+metadata:
+  name: ${CLUSTER_NAME}-mp-1
+spec:
+  minSize: 1
+  maxSize: 4
+  awsLaunchTemplate:
+    instanceType: "${AWS_CONTROL_PLANE_MACHINE_TYPE}"
+    iamInstanceProfile: "nodes.cluster-api-provider-aws.sigs.k8s.io"
+    sshKeyName: "${AWS_SSH_KEY_NAME}"
+    spotMarketOptions:
```

Comment:
Should this be:

```yaml
spotMarketOptions: {}
```

Reply:
I referred the existing test where we have specified nil maxPrice

> spotMarketOptions:
>      maxPrice: ""

Reply:
Can this be relapced with 
```yaml
spotMarketOptions: {}
```

Reply:
I think so the check is for non-nil options.

Reply:
@sedefsavas so shall we keep it as is?